packages/datadog-plugin-router/test/http2.spec.js

```diff
@@ -0,0 +1,167 @@
+'use strict'
```

Comment:
Can both tests be merged with just a parameter of which HTTP module to use? Otherwise this is a lot of duplication.

Reply:
Yeah, should be able to do something like:

const tests = [
  { name: 'http', createServer () { /* ... */ } },
  { name: 'http2', createServer () { /* ... */ } },
]

for (const { name, createServer } of tests) {
  describe(name, () => {
    // use createServer wherever needed, and add the tests...
  })
}

Reply:
I don't think this is a good idea.

The actual implementation of the tests are slightly different, so there'd end up being a lot of if statement in 'em.

For http2, fetch-h2 is being used as the client, as axios doesn't support http2. And the URLs being passed to fetch-h2 start with http2://, so that it knows we're dealing with a http2 clearText request

so even though the test names are the same, and the expectations are similar, how the data is fetched is different.

Reply:
Why not just use an HTTP client that supports both? Then this becomes a non-issue and both tests can be merged.